FlashCap.Core/Devices/DirectShowDevice.cs

```diff
@@ -312,6 +313,106 @@ protected override Task OnStopAsync(CancellationToken ct)
         }
     }
 
+    public override int GetPropertyValue(VideoProcessingAmplifierProperty property)
+    {
+        if (Properties.TryGetValue(property, out CaptureDeviceProperty _)
+            && graphBuilder != null
+            && captureGraphBuilder != null)
+        {
+            graphBuilder.FindFilterByName("Capture source", out NativeMethods_DirectShow.IBaseFilter? captureSourceFilter);
```

Comment:
in my application (wpf), it's working fine, but in the the sample I've created based on OneShot, I'm getting this 
`Unable to cast COM object of type 'System.__ComObject' to interface type 'IBaseFilter'.`

Reply:
@kekyo I was currently stuck here, any idea why I'm experiencing this?

Reply:
OK, I will check it.

Reply:
@ldlac In this source file, there are 3 places that use `FindFilterByName`, do all of them throw the same exception? In my environment, the `FindFilterByName` in `DisplayPropertyPage_CaptureFilter` also threw an exception.
If so, there may be something wrong with `FindFilterByName` (I don't know the cause yet). If not, then we need to suspect another cause...

Reply:
@ldlac I debugged the code and found a problem with the added property page display and the property getter/setter accesses ignoring the COM apartments, so I fixed it.  

472f25d9966cb04c638c12a0902fbbceb660b544  (I think the `develop` was badly positioned and that caused the weird merge (My mistake), so I fixed it in another branch. `develop` has been repositioned, so please rebase it.)

FlashCap has been modified to initialize another thread as a single-threaded apartment and access DS objects only on it, as past implementation experience has shown that accessing the calling COM apartment as it is causes many frame drops The following is a list of the DS objects that can be accessed beyond the apartment.

When accessing the DS object beyond the apartment, the standard COM way is to marshal the interface, but it is tedious to do this in .NET, so we use `SynchronizationContext` to send messages. See `IndeependentSingleApartmentContext.InvokeAsync()`.

In the environment at hand, this modification allows the capture filter property page to be displayed.
I have not tested the getter and setter, but the modification should give you an idea of what to do.

Reply:
Thanks for this commit, really helped!

The sample is working fine, I'll need to handle `flags` 

![image](https://user-images.githubusercontent.com/12978080/233795451-057b9209-d841-42f0-bc9d-bd79eb1a2ed6.png)